### PR TITLE
Rewrite renderers according to Issue #1

### PR DESCRIPTION
**Pull request by matiit**
PR for Issue #1

### CODE REVIEW

**Comment by danhanly** [comment]
Pretty nice; I'll give this a spin later tonight.

File: .notifly.yml.dist [hunk 1]
```diff
@@ -1,6 +1,3 @@
-renderers:
-  error: \Notifly\Renderer\Error
-  warning: \Notifly\Renderer\Warning
-  info: \Notifly\Renderer\Info
-  success: \Notifly\Renderer\Success
-driver: \Notifly\Storage\SessionDriver
\ No newline at end of file
+renderer:
+  html: \Notifly\Renderer\HtmlRenderer
```
Comment (danhanly) [inline .notifly.yml.dist:2]: Similar to how drivers are set up, can we not simply add `renderer: \Notifly\Renderer\HtmlRenderer`
Apologies if I've misunderstood the intention here.

In-fact, looking at the config below, that seems to be how it's set up already; just needs reflecting here.

File: src/Notifly/Exception/MessageTypeUnsupported.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Notifly\Exception;
+
+class MessageTypeUnsupported extends \Exception
```
Comment (danhanly) [inline src/Notifly/Exception/MessageTypeUnsupported.php:5]: Can you prefix this class name with `Notifly`
i.e. `NotiflyMessageTypeUnsupported`

File: .notifly.yml.dist (hunk 1 above)
Comment (matiit) [inline .notifly.yml.dist:2]: I agree. Will change that.

File: src/Notifly/Renderer/HtmlRenderer.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * Create and return html string.
+     *
+     * @param array $notifications
+     * @param string $type
+     * @return string
+     */
+    private function createHtmlByType(array $notifications, $type)
```
Comment (danhanly) [inline src/Notifly/Renderer/HtmlRenderer.php:125]: This duplicates functionality provided by the `render` method. Can you simply call `createHtmlByType` from the render method?

File: phpunit-bootstrap.php [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?php
```
Comment (danhanly) [inline phpunit-bootstrap.php:1]: Given that this is more of a config file for the test suite, can you add this file to the config directory?

**Comment by danhanly** [comment]
I've added Travis to the main branch now, and included within it support for PHP 5.4 (trying to widen the market of potential projects) - can you make this branch up to date with that? Then we'll be good for first release.

**Comment by danhanly** [comment]
Thinking about it now; we perhaps need to think about how the render methods are called. They'd be called against the Notifly object, which would pass through to the configured renderer. The `renderError` etc. methods become pointless in this case.

``` php
$notifly = new Notifly();
$notifly->render($class, $bucket)
```

Then the `HtmlRenderer` simply needs a `render` method and that's it.

**Comment by danhanly** [comment]
I ended up making these changes, and added some other stuff.

**Pull request closed without merging**